AppSyncRealTimeClient/Websocket/Starscream/StarscreamAdapter.swift

```diff
@@ -9,22 +9,41 @@ import Foundation
 import Starscream
 
 public class StarscreamAdapter: AppSyncWebsocketProvider {
-    public init() {
-        // Do nothing
-    }
-
-    private let serialQueue = DispatchQueue(label: "com.amazonaws.StarscreamAdapter.serialQueue")
+    let serialQueue: DispatchQueue
+    private let callbackQueue: DispatchQueue
 
     var socket: WebSocket?
     weak var delegate: AppSyncWebsocketDelegate?
 
+    // swiftlint:disable:next identifier_name
+    var _isConnected: Bool
+    public var isConnected: Bool {
+        serialQueue.sync {
+            _isConnected
+        }
+    }
+
+    public init() {
+        let serialQueue = DispatchQueue(label: "com.amazonaws.StarscreamAdapter.serialQueue")
+        let callbackQueue = DispatchQueue(
+            label: "com.amazonaws.StarscreamAdapter.callBack",
+            target: serialQueue
```

Comment:
ref https://stackoverflow.com/questions/59828733/how-current-queue-dispatch-queue-and-target-queue-communicates-with-each-other